### PR TITLE
[JBPM-9649] - Adding integration test for checking parent/subprocess relationships

### PR DESCRIPTION
**Pull request by afalhambra**
**[JBPM-9649](https://issues.redhat.com/browse/JBPM-9649)**: Adding integration test for checking parent/subprocess relationships

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/admin/ProcessInstanceMigrationIntegrationTest.java [hunk 1]
```diff
@@ -285,92 +287,223 @@ public void testUpgradeProcessInstancesWithNodeMapping() {
...
-                assertEquals(CONTAINER_ID, processInstance.getContainerId());
-                assertEquals(SUBPROCESS_ID, processInstance.getProcessId());
-                assertEquals(processParentInstanceId, processInstance.getParentId());
-            }
+            List<ProcessInstance> childrenProcessInstance = processClient.findProcessInstancesByParent(CONTAINER_ID, processParentInstanceId, 0, 10);
+            assertEquals(1, childrenProcessInstance.size());
+            assertProcessInstancesInfo(childrenProcessInstance, "subprocessCallingAnotherSubProcess", "1.0",
+                                       CONTAINER_ID, SUBPROCESS_CALLING_ANOTHER_SUBPROCESS_ID, processParentInstanceId);
+
+            List<ProcessInstance> childrenSubProcessInstance = processClient.findProcessInstancesByParent(CONTAINER_ID, childrenProcessInstance.get(0).getId(), 0, 10);
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/admin/ProcessInstanceMigrationIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/admin/ProcessInstanceMigrationIntegrationTest.java [hunk 2]
```diff
@@ -285,92 +287,223 @@ public void testUpgradeProcessInstancesWithNodeMapping() {
...
+        }
+
+        tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+        logger.info("children tasks instances from {} fetched", CONTAINER_ID_2);
+        assertEquals(2, tasks.size());
+
+        for (TaskSummary task : tasks) {
+            assertTaskInfo(task, "Approve", CONTAINER_ID_2, SUBPROCESS_ID_2);
+            taskClient.completeAutoProgress(CONTAINER_ID_2, task.getId(), USER_YODA, null);
+        }
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/admin/ProcessInstanceMigrationIntegrationTest.java]: done

**Comment by afalhambra** [comment]
thanks for your review and suggestions @gmunozfe

**Review by gmunozfe — APPROVED** [review]
Looks good to me, top-notch work @afalhambra !

**Comment by afalhambra** [comment]
@elguardian could you please merge?

**Pull request merged**